protocol/signalfx/log_hecv1_json.go

```diff
@@ -0,0 +1,66 @@
+package signalfx
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mailru/easyjson"
+	"github.com/signalfx/golib/v3/log"
+	"github.com/signalfx/golib/v3/logsink"
+	signalfxformat "github.com/signalfx/ingest-protocols/protocol/signalfx/format"
+)
+
+const (
+	// resources
+	host   = "host.hostname"
+	source = "service.name"
+	// attributes
+	sourceType = "source.type"
+	index      = "index"
+)
+
+// JSONDecoderHECV1 implements sink decoder for hecv1 splunk log events
+type JSONDecoderHECV1 struct {
+	Sink   logsink.Sink
+	Logger log.Logger
+}
+
+// Read implements interface method for decoding incoming json hecv1 format and convert it into golib/logsink.Log format
+func (decoder *JSONDecoderHECV1) Read(ctx context.Context, req *http.Request) (err error) {
+	var logHECV1List signalfxformat.JSONLogHECV1List
+	if err = easyjson.UnmarshalFromReader(req.Body, &logHECV1List); err == nil {
+
+		logs := make(logsink.Logs, len(logHECV1List))
+		for idx, lEntry := range logHECV1List {
+			e := &logsink.Log{
```

Comment:
Is it possible for `lEntry` to be nil or it is guaranteed to be present after unmarshaling?

Reply:
@jgheewala unit test for this ☝️

Reply:
looking at easyjson unmarshalling code we may get nil if the scanning of incoming data yields null. I will fix the nil part

Reply:
added test case to handle the nil case